zerver/tests/test_auth_backends.py

```diff
@@ -887,7 +889,7 @@ def test_social_auth_deactivated_user(self) -> None:
         # We expect to go through the "choose email" screen here,
         # because there won't be an existing user account we can
         # auto-select for the user.
-        with mock.patch('zproject.backends.logging.info') as m:
+        with mock.patch.object(self.logger, 'info') as m:
             result = self.social_auth_test(account_data_dict,
                                            expect_choose_email_screen=True,
                                            subdomain='zulip')
```

Comment:
I suppose this works, but maybe it's worth looking at whether there's a better interface for mocking a logger during tests?

Reply:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs this seems like an interesting option to explore. If it works well, perhaps we should even consider using it to replace all of the ugly logger mocks we have been doing across our tests?

Reply:
FYI @chdinesh1089

Reply:
Yeah, that seems like the right way to do this.  

@chdinesh1089 would you be up for redoing the testing part of this using that feature?  If we like the outcome, we can then do it to the whole codebase.

Reply:
Yes, did the testing part again.